read_buffer/src/lib.rs

```diff
@@ -199,6 +200,18 @@ impl Database {
             Some(partition) => {
                 let mut chunks = vec![];
                 for chunk_id in chunk_ids {
+                    let chunk = partition
+                        .chunks
+                        .get(chunk_id)
+                        .ok_or_else(|| Error::ChunkNotFound { id: *chunk_id })?;
```

Comment:
You can use the Snafu `context` thing with `Option`s too if you want: 

```suggestion
                        .context(ChunkNotFound { id: chunk_id })?;
```

Though you have to add a `use snafu::OptionExt`